amplify/ts-code/src/db/ScheduleTable.ts

```diff
@@ -0,0 +1,131 @@
+import { SCHEDULE_TABLE, ScheduleSchema, ITEMS_TABLE, ItemsSchema } from "./Schemas";
+import { DBClient } from "../injection/db/DBClient"
+import { DocumentClient } from "aws-sdk/clients/dynamodb"
+
+export class ScheduleTable {
+    private readonly client: DBClient
+
+    public constructor(client: DBClient) {
+        this.client = client
+    }
+
+    /**
+     * Adds item to inventory
+     */
+    public create(
+        id: string,
+        borrower: string,
+        itemId: string,
+        startTime: string,
+        endTime: string,
+        notes: string,
+    ): Promise<DocumentClient.PutItemOutput> {
+        return this.get(id)
+            .then((entry: ScheduleSchema) => {
+                if (entry) {
+                    throw Error(`Schedule ID ${id} is not unique.`)
+                } else {
+                    const itemsParams: DocumentClient.UpdateItemInput = {
+                        TableName: ITEMS_TABLE,
+                        Key: {
+                            "id": itemId
+                        },
+                        UpdateExpression: "SET #key = list_append(#key, :val)",
+                        ExpressionAttributeNames: {
+                            "#key": "schedule"
+                        },
+                        ExpressionAttributeValues: {
+                            ":val": [id]
+                        }
+                    }
+
+                    const reservation: ScheduleSchema = {
+                        id: id,
+                        borrower: borrower,
+                        itemId: itemId,
+                        startTime: startTime,
+                        endTime: endTime,
+                        notes: notes
+                    }
+
+                    const indexParams: DocumentClient.PutItemInput = {
+                        TableName: SCHEDULE_TABLE,
+                        Item: reservation
+                    }
+
+                    return this.client.update(itemsParams)
+                        .then(() => this.client.put(indexParams))
```

Comment:
I think you want to do a `put` before the `update`. If you're unable to put the object into the schedule table, then you don't want to append to the item table prematurely. Meanwhile, a orphaned `schedule` object, where there's no reference from item -> schedule is not the worst thing.

Reply:
Gotcha!